Week 6, survey-project, Petra Söderström

Petrasoderstrom1612 opened the pull request:
Link to my deployed project:

https://star-survey.netlify.app/

juliaostedt commented on code/src/components/UserLocation.js:
```diff
@@ -0,0 +1,25 @@
+
+  return (
+    <>
+      <p className="star-shop" tabIndex="0">What is your nearest physical <span className="star">Star shop</span>⭐? </p>
+      <p className="user-p" tabIndex="0">Select one of the options below</p>
```
Suggestion would be to use h2-h6 instead of a p-tag here, that way you do not need to add the tabindex. Here's a good article on tabindexes, https://webaim.org/techniques/keyboard/tabindex.

juliaostedt commented on code/src/index.css:
```diff
@@ -1,13 +1,410 @@
+  background-size: cover;
+    background-repeat: no-repeat;
+    background-position: center; 
+    height: 100vh;
```
Seem to be a few too many indentations here. And height is listed twice. :)

juliaostedt commented on code/src/index.css:
```diff
@@ -1,13 +1,410 @@
+  } 
+
+  .subscribe-button:hover {
+    font-size: 35px;
+    transition: all 0.1s ease-out 0s;
```
Suggestion to try and see if the `transform: scale()` property works instead of increasing the fontsize, I believe this would help to make the other parts of the page not to move when hovering. Here's a good article on the scale property: https://css-tricks.com/almanac/properties/s/scale/

juliaostedt commented on code/src/App.js:
```diff
@@ -1,9 +1,148 @@
+    }
+  };
+
+  const handleClearButtonClick = () => {
```
Great that you added a clear-button!

juliaostedt commented on code/src/components/UserMail.js:
```diff
@@ -0,0 +1,16 @@
+  return (
+    <div>
+      <p className="user-p" tabIndex="0">What is your email address?</p>
+      <input value={usermail} className="user-input" type="email" placeholder="email address" required="required" onChange={handleMailInputChange} />
```
I'm not sure why, but it doesn't seem like the `type="email"` is working properly. I'm able to go to the next questions without adding a valid email address with a "@" and ".com". Perhaps worth looking into?

juliaostedt commented on code/src/components/UserName.js:
```diff
@@ -0,0 +1,17 @@
+    setUsername(event.target.value);
+  };
+  return (
+    <div>
+      <p className="user-p" tabIndex="0"> Let's start with your first name.</p>
+      <input type="text" className="user-input" tabIndex="0" value={username} onChange={handleUserNameInputChange} placeholder="name" name="name" required />
+    </div>
```
It seems like the required function isn't working properly, I wonder if you need to use `form` instead of a `div` to wrap your inputs and question-texts?

juliaostedt reviewed (commented):
Great job, it's a nice looking survey!! 👏 I've added a few comments with suggestions and would also recommend you to move your images/icons into the same folders, right now they are spread into three different ones (images/assets/public).

Petrasoderstrom1612 commented:
<!--
/* Font Definitions */
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
@font-face
	{font-family:"Segoe UI Emoji";
	panose-1:2 11 5 2 4 2 4 2 2 3;}
/* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin:0cm;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	text-decoration:underline;}
code
	{mso-style-priority:99;
	font-family:"Courier New";}
pre
	{mso-style-priority:99;
	mso-style-link:"HTML - f\00F6rformaterad Char";
	margin:0cm;
	font-size:10.0pt;
	font-family:"Courier New";}
span.HTML-frformateradChar
	{mso-style-name:"HTML - f\00F6rformaterad Char";
	mso-style-priority:99;
	mso-style-link:"HTML - f\00F6rformaterad";
	font-family:"Courier New";}
.MsoChpDefault
	{mso-style-type:export-only;}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:70.85pt 70.85pt 70.85pt 70.85pt;}
div.WordSection1
	{page:WordSection1;}
-->Hej! Osäker om jag har tackat så skriver en gång till. Superbra feedback! Och de här länkarna är topnotch! Tack Julia 😊 Skickades från E-post för Windows Från: Julia ÖstedtSkickat: den 3 oktober 2022 10:41Till: Technigo/project-surveyKopia: Petrasoderstrom1612; AuthorÄmne: Re: [Technigo/project-survey] Week 6, survey-project, Petra Söderström (PR ***@***.*** commented on this pull request.Great job, it's a nice looking survey!! 👏 I've added a few comments with suggestions and would also recommend you to move your images/icons into the same folders, right now they are spread into three different ones (images/assets/public).In code/src/components/UserLocation.js:> +      <p className="star-shop" tabIndex="0">What is your nearest physical <span className="star">Star shop</span>⭐? </p> +      <p className="user-p" tabIndex="0">Select one of the options below</p> Suggestion would be to use h2-h6 instead of a p-tag here, that way you do not need to add the tabindex. Here's a good article on tabindexes, https://webaim.org/techniques/keyboard/tabindex.In code/src/index.css:> +  justify-content: center; +  align-items: center; +  text-align: center; +  height: 100vh; +  position: fixed;  +  background-color: black; +  overflow-y: scroll; +  top: 0; right: 0; bottom: 0; left: 0; +  background-image: linear-gradient( +      rgba(11, 11, 11, 0.65), +      rgba(11, 11, 11, 0.65) +    ), url(./assets/background.jpg); +  background-size: cover; +    background-repeat: no-repeat; +    background-position: center;  +    height: 100vh; Seem to be a few too many indentations here. And height is listed twice. :)In code/src/index.css:> +    font-size: 35px; +    transition: all 0.1s ease-out 0s; Suggestion to try and see if the transform: scale() property works instead of increasing the fontsize, I believe this would help to make the other parts of the page not to move when hovering. Here's a good article on the scale property: https://css-tricks.com/almanac/properties/s/scale/In code/src/App.js:> +  const [name, setName] = useState(''); +  const [lastname, setLastname] = useState(''); +  const [mail, setMail] = useState(''); +  const [phone, setPhone] = useState(''); +  const [location, setLocation] = useState(''); +  const [age, setAge] = useState(''); +  const [frequency, setFrequency] = useState([]); +  const handleCounterButtonClick = (shouldIncrease) => { +    if (shouldIncrease && counter < 8) { +      setCounter(counter + 1); +    } else if (!shouldIncrease && counter > 0) { +      setCounter(counter - 1); +    } +  }; + +  const handleClearButtonClick = () => { Great that you added a clear-button!In code/src/components/UserMail.js:> @@ -0,0 +1,16 @@ +/* eslint-disable jsx-a11y/no-noninteractive-tabindex */ +import React from 'react'; + +const UserMail = ({ usermail, setUsermail }) => { +  const handleMailInputChange = (event) => { +    setUsermail(event.target.value); +  }; +  return ( +    <div> +      <p className="user-p" tabIndex="0">What is your email address?</p> +      <input value={usermail} className="user-input" type="email" placeholder="email address" required="required" onChange={handleMailInputChange} /> I'm not sure why, but it doesn't seem like the type="email" is working properly. I'm able to go to the next questions without adding a valid email address with a "@" and ".com". Perhaps worth looking into?In code/src/components/UserName.js:> +    <div> +      <p className="user-p" tabIndex="0"> Let's start with your first name.</p> +      <input type="text" className="user-input" tabIndex="0" value={username} onChange={handleUserNameInputChange} placeholder="name" name="name" required /> +    </div> It seems like the required function isn't working properly, I wonder if you need to use form instead of a div to wrap your inputs and question-texts?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***> 
